src/components/ScrollShadows.tsx

```diff
@@ -0,0 +1,103 @@
+import { useLayoutEffect } from "@react-aria/utils";
+import { ReactNode, useCallback, useRef, useState } from "react";
+import { Css, Palette, Properties } from "src";
+
+interface ScrollShadowsProps {
+  children: ReactNode;
+  /** Allows for styling the container */
+  xss?: Properties;
+  /** Set to true if the container scrolls horizontally */
+  horizontal?: boolean;
+  /** Defines the background color for the shadows */
+  bgColor?: Palette;
+}
+export function ScrollShadows(props: ScrollShadowsProps) {
+  const { children, xss, horizontal = false, bgColor = Palette.White } = props;
+  const { height = "auto", width = "auto" } = xss ?? {};
+  // This is admittedly extremely hacky. It expects the background color to be in the format "rgba(255, 255, 255, 1)".
+  // If we ever change how we define our color palette in Beam, then this will break and will need to be fixed.
+  const transparentBgColor = bgColor.replace(/,1\)$/, ",0)");
+
+  const startShadowStyles = !horizontal ? Css.top0.left0.right0.hPx(40).$ : Css.left0.top0.bottom0.wPx(25).$;
+  const endShadowStyles = !horizontal ? Css.bottom0.left0.right0.hPx(40).$ : Css.right0.top0.bottom0.wPx(25).$;
+  const startGradientDeg = !horizontal ? 180 : 90;
+  const endGradientDeg = !horizontal ? 0 : 270;
+
+  const [scrollProps, setScrollProps] = useState<ScrollProps>({
+    scrollTop: 0,
+    scrollHeight: 0,
+    clientHeight: 0,
+    clientWidth: 0,
+    scrollLeft: 0,
+    scrollWidth: 0,
+  });
+  const scrollRef = useRef<HTMLDivElement>(null);
+
+  const updateScrollProps = useCallback((el: HTMLDivElement) => {
+    const { scrollTop, scrollHeight, clientHeight, scrollWidth, scrollLeft, clientWidth } = el;
+    setScrollProps({ scrollTop, scrollHeight, clientHeight, scrollWidth, scrollLeft, clientWidth });
+  }, []);
+
+  // Initially set the state dimensions on render to put the shadows in the correct position
+  useLayoutEffect(() => {
+    scrollRef.current && updateScrollProps(scrollRef.current);
+  }, []);
+
+  const { scrollTop, scrollHeight, clientHeight, scrollWidth, scrollLeft, clientWidth } = scrollProps;
+  const showStartShadow = !horizontal ? scrollTop > 0 : scrollLeft > 0;
+  const showEndShadow = !horizontal ? scrollTop + clientHeight < scrollHeight : scrollLeft + clientWidth < scrollWidth;
```

Comment:
Praise: Nice showstart/end variables! Makes this pretty readable.

...

Could potentially do:

```
const start = horizontal ? scrollLeft : scrollTop;
const len = horizontal ? clientWidth : clientHeight;
const end = horizontal ? scrollWidth : scrollHeight;
const showStartShadow = start > 0;
const showEndShadow = start + len < end;
```

...could maybe even work ^ into the `useState`, like instead of storing all 6 in the `useState`, just store the three `start`/`len`/`end`, and do the `horizontal`-yes/no check during `updateScrollProps`.

Reply:
Yea, I like that idea. I will update the state.

Reply:
I think I might change this to using two states: `showStartShadow` and `showEndShadow`. That might make this a bit cleaner.